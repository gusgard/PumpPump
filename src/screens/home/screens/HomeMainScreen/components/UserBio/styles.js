import { StyleSheet } from 'react-native';

import { horizontal, vertical, colors, fontSize } from '@themes';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginHorizontal: horizontal.small,
    marginVertical: vertical.small,
  },
  texts: {
    // borderWidth: 1,
    flex: 1,
    marginLeft: horizontal.small,
  },
  name: {
    // borderWidth: 1,
    flex: 1,
    fontSize: fontSize.large,
    fontWeight: 'bold',
  },
  bio: {
    borderWidth: 1,
    // justifyContent: 'flex-end',
    // alignSelf: 'flex-end',
    // alignItems: 'center',
    // flex: 1,
    fontSize: fontSize.medium,
  },
  highlighted: {
    fontWeight: 'bold',
    color: colors.primaryDark,
  },
});
