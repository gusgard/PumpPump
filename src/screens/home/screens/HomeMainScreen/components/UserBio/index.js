import React, { PureComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import { Thumbnail } from '@components';

import styles from './styles';

export default class UserBio extends PureComponent {
  static propTypes = {
    name: PropTypes.string.isRequired,
    bio: PropTypes.array.isRequired,
    picture: PropTypes.object.isRequired,
  };

  render() {
    const { name, bio, picture } = this.props;
    return (
      <View style={styles.container}>
        <Thumbnail source={picture} />
        <View style={styles.texts}>
          {/* TOP */}
          <Text style={styles.name}>{name}</Text>
          {/* BOTTOM */}
          <Text numberOfLines={3} style={styles.bio}>
            {bio.map(
              (text, index) =>
                text.highlighted ? (
                  <Text key={index} style={styles.highlighted}>
                    {text.value}
                  </Text>
                ) : (
                  <Text key={index}>{text.value}</Text>
                ),
            )}
          </Text>
        </View>
      </View>
    );
  }
}
