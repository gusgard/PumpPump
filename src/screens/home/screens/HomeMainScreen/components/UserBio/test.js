import { shallow } from 'enzyme';
import React from 'react';

import UserBio from './';

const user = {
  bio: [],
  name: '',
  thumbnail: {},
};

it('renders correctly', () => {
  const wrapper = shallow(
    <UserBio name={user.name} bio={user.bio} picture={user.thumbnail} />,
  );
  expect(wrapper).toMatchSnapshot();
});
