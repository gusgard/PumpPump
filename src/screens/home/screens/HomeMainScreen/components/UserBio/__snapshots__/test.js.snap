// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginHorizontal": 28.125,
      "marginVertical": 50.025,
    }
  }
>
  <Thumbnail
    size={100}
    source={Object {}}
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 28.125,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Object {
          "borderWidth": 1,
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;
